### DISCUSSION TITLE
Meeting: January 11th, 2021

### DISCUSSION THREAD

**Discussion by gkiar** [yohanchatelain/sigdigits#7] [General]
Purpose: Discuss the motivation and plan for the project, including use cases, development plan, etc.

Add notes below 🙂

**Comment by glatard** [comment]
* @pablooliveira has code that should be compared to the current implementation

**Reply by gkiar** [reply]
We should compare all previously used implementations. I know both @ali4006 and I have one which our papers have used, so we should see if the interpretations of results changes based on changing implementations and issue follow-ups to papers as necessary, accordingly.

**Comment by gkiar** [comment]
Use cases:
- Evaluate significance (many sub-use cases)
- Compute known confidence interval on value

**Comment by pablooliveira** [comment]
Default values for p and 1-alpha, could be .9 and .95 respectively, which requires 29 samples in the general case. It's what we have been using here by default.

**Comment by gkiar** [comment]
License: @glatard  MIT preferred over GPL3 since people tend to avoid contaminating licenses sometimes.

**Comment by gkiar** [comment]
- We should clearly document the correct formula based on the data type and properties of the signal (i.e. meaningless zero or meaningfully zero-centered signals). A choose-your-adventure type doc page would be great.

**Comment by gkiar** [comment]
We should follow best-practices for python packaging (there is an example we could follow, [here](https://github.com/uwescience/shablona))

**Comment by gkiar** [comment]
- What do we do when std is zero? We could do absolute difference @yohanchatelain 
- Should we cap/bound the precision (i.e. we could obtain an infinite value in some cases with low std, do we want to cap at number of bits?)

**Comment by gkiar** [comment]
- How do we determine the significance datatype to use? i.e. with 16bit inputs and 32bit outputs for the pipeline, which data type do we use for computing significance?

**Comment by gkiar** [comment]
We'll move this and gkiar/fuzzy to verificarlo org

**Reply by pablooliveira** [reply]
You should have enough privileges to create new packages under the `verificarlo` organisation; if that is not the case please tell us and we will update the permissions.

**Reply by yohanchatelain** [reply]
Thanks! Should we start the project from scratch or imported it from this repo?

**Comment by gkiar** [comment]
- We should give the users lots of warnings for things like "getting close to zero, consider other method", "not enough samples for this method", etc...

**Comment by yohanchatelain** [comment]
Move this discussion into https://github.com/verificarlo/significantdigits/discussions
